### PR TITLE
001 Script for requirement [GENIVI] PerformAudioPassThru: SDL must suppor…

### PR DESCRIPTION
**Pull request by maria-mih**
…t new audioPassThruIcon parameter
001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua

All common functions used available at #703
Testing coverage available at #643

@istoimenova Please review

### CODE REVIEW

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---------------------------------------------------------------------------------------------
+-- Requirements summary:
+-- [PerformAudioPassThru] SDL must transfer request to HMI in case of valid "audioPassThruIcon" param
+-- [HMI API] UI.PerformAudioPassThru request/response
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, please reference non-functional requirement for TTS:
[HMI API] TTS.Speak request/response

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+--1. Used preconditions
+--1. All params used in PerformAudioPassThru_request are present and within bounds
+--2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, other params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
+--2.2 SDL sends TTS.Speak to HMI
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
+--2.4 HMI sends TTS.Speak (SUCCESS) to SDL
+-- Expected result:
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih , expected result should be:
SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
SDL sends TTS.Speak to HMI

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+-- as well as another related to request valid params
+-- SDL must transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--1. Used preconditions
+--1. All params used in PerformAudioPassThru_request are present and within bounds
+--2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, other params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
+--2.2 SDL sends TTS.Speak to HMI
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, points 2.3-2.4 are redundant in description.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_AllParameters_SUCCESS()
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, as the idea of atomic script is to show what exactly is happening, will be better to declare the TestStep in script, not to call common function.
Calling of common functions is useful as preconditions/postconditions.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih , please be sure that RPC PerformAudioPassThru is allowed by policy.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+commonFunctions:newTestCasesGroup("Preconditions")
+
+commonSteps:PutFile("Precondition_PutFile_With_Icon", "icon.png")
+
+function Test:Precondition_Check_audioPassThruIcon_Existence()
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih ,please use parameters of function on the same way

**Comment by istoimenova** [comment]
@maria-mih, please reference script in the TestSet.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 1 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @istoimenova 
Requirement is added

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @istoimenova 
Description is updated

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @istoimenova 
Description is updated

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 6 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @istoimenova 
Function is aligned as expected

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 5 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @istoimenova 
Please check in section --[[ General Precondition before ATF start ]] , function `testCasesForPolicyTable:precondition_updatePolicy_AllowFunctionInHmiLeves` is added to allow RPC.
Also in section --[[ Postconditions ]] appropriate function for restoration of default settings is added.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @istoimenova 
Function has been used in full.

**Comment by maria-mih** [comment]
@istoimenova 
Script has been updated after first review iteration. Please check on commit f61e2efa20232bbf6c8bf54e4c181b1d627b624a

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 2 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 6 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, ACCEPTED.

Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 5 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, ACCEPTED

File: test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua (hunk 4 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/001_ATF_P_PerformAudioPassThru_SUCCESS_All_Params.lua]: @maria-mih, ACCEPTED.

**Comment by istoimenova** [comment]
@maria-mih, review iteration is finished.

**Pull request merged**